blueprints/data-solutions/vertex-mlops/main.tf

```diff
@@ -0,0 +1,267 @@
+locals {
+  groups = {
+    for k, v in var.groups : k => "${v}@${var.organization_domain}"
+  }
+  groups_iam = {
+    for k, v in local.groups : k => "group:${v}"
+  }
+
+  project_groups_iam = {
+    (local.groups.data-scientists) = [
+      "roles/aiplatform.admin",
+      "roles/artifactregistry.admin",
+      "roles/bigquery.dataEditor",
+      "roles/bigquery.jobUser",
+      "roles/bigquery.user",
+      "roles/cloudbuild.builds.editor",
+      "roles/cloudfunctions.developer",
+      "roles/dataflow.developer",
+      "roles/dataflow.worker",
+      "roles/iam.serviceAccountUser",
+      "roles/logging.logWriter",
+      "roles/logging.viewer",
+      "roles/notebooks.admin",
+      "roles/pubsub.editor",
+      "roles/serviceusage.serviceUsageConsumer",
+      "roles/storage.admin"
+    ],
+    (local.groups.ml-engineers) = [
+      "roles/aiplatform.admin",
+      "roles/artifactregistry.admin",
+      "roles/bigquery.dataEditor",
+      "roles/bigquery.jobUser",
+      "roles/bigquery.user",
+      "roles/dataflow.developer",
+      "roles/dataflow.worker",
+      "roles/iam.serviceAccountUser",
+      "roles/logging.logWriter",
+      "roles/logging.viewer",
+      "roles/serviceusage.serviceUsageConsumer",
+      "roles/storage.admin"
+    ]
+  }
+
+  # internal structure for Shared VPC service project IAM bindings
+  _vpc_subnet_bindings = (
+    local.vpc.subnets_iam == null || local.vpc.host_project == null
+    ? []
+    : flatten([
+      for subnet, members in local.vpc.subnets_iam : [
+        for member in members : {
+          region = split("/", subnet)[0]
+          subnet = split("/", subnet)[1]
+          member = member
+        }
+      ]
+    ])
+  )
+
+  # structures for Shared VPC resources in host project
+  vpc = coalesce(var.vpc, {
+    host_project = null, gke_setup = null, subnets_iam = null
+  })
+  vpc_cloudservices = (
+    local.vpc_gke_service_agent
+  )
+  vpc_gke_security_admin = coalesce(
+    try(local.vpc.gke_setup.enable_security_admin, null), false
+  )
+  vpc_gke_service_agent = coalesce(
+    try(local.vpc.gke_setup.enable_host_service_agent, null), false
+  )
+  vpc_subnet_bindings = {
+    for binding in local._vpc_subnet_bindings :
+    "${binding.subnet}:${binding.member}" => binding
+  }
+
+}
+
+
+module "gcs-bucket" {
+  count         = var.bucket_name == null ? 0 : 1
+  source        = "../../../modules/gcs"
+  project_id    = module.project.project_id
+  name          = var.bucket_name
+  prefix        = var.prefix
+  location      = var.region
+  storage_class = "REGIONAL"
+  versioning    = false
+}
+
+module "gcs-bucket-cloudbuild" {
+  # Default bucket for Cloud Build to prevent error: "'us' violates constraint ‘constraints/gcp.resourceLocations’"
+  # https://stackoverflow.com/questions/53206667/cloud-build-fails-with-resource-location-constraint
+  source        = "../../../modules/gcs"
+  project_id    = module.project.project_id
+  name          = "${var.project_id}_cloudbuild"
```

Comment:
I would rely on the prefix and remove the project id.

Reply:
Sorry, I don't get it. What do you mean?

Reply:
I would remove the `project_id` in the `name` and rely on the combination of prefix + name. Similar to all other blueprints, example: https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/20826d0ef51aa575c247b1619e9d0a1839300804/blueprints/data-solutions/gcs-to-bq-with-least-privileges/datastorage.tf#L15

Reply:
Got it, thanks!